code/modules/mob/living/death.dm

```diff
@@ -49,8 +49,14 @@
 	timeofdeath = world.time
 	tod = worldtime2text()
 	var/turf/T = get_turf(src)
+	var/area/A = get_area(T)
+	if(src.mind.special_role == "Cult Master")

```

Comment:
Is there example I can use? I don't know of any status effect besides diseases/mutations that are permanent and disable on death.